api/v1alpha1/etcd_types.go

```diff
@@ -393,3 +393,95 @@ type EtcdList struct {
 	metav1.ListMeta `json:"metadata,omitempty"`
 	Items           []Etcd `json:"items"`
 }
+
+// LastOperationState is a string alias.
+type LastOperationState string
+
+const (
+	// LastOperationStateActive indicates that an operation is ongoing.
+	LastOperationStateActive LastOperationState = "Active"
+	// LastOperationStateSucceeded indicates that an operation has completed successfully.
+	LastOperationStateSucceeded LastOperationState = "Succeeded"
+	// LastOperationStateFailed indicates that an operation is completed with errors and won't be retried.
+	LastOperationStateFailed LastOperationState = "Failed"
+)
+
+// LastOperation indicates the type and the state of the last operation, along with a description
+// message and a progress indicator.
+type LastOperation struct {
+	// A human readable message indicating details about the last operation.
+	Description string `json:"description" protobuf:"bytes,1,opt,name=description"`
+	// Last time the operation state transitioned from one to another.
+	LastUpdateTime metav1.Time `json:"lastUpdateTime" protobuf:"bytes,2,opt,name=lastUpdateTime"`
+	// State shows the status of the last operation, one of Aborted, Processing, Succeeded, Error, Failed.
+	State LastOperationState `json:"state" protobuf:"bytes,4,opt,name=state,casttype=LastOperationState"`
+}
+
+// EtcdCopyBackupsTaskSpec defines the parameters for the copy operation
+type EtcdCopyBackupsTaskSpec struct {
+	// SourceStore defines the specification of object store provider for storing backups.
+	// +required
+	SourceStore *StoreSpec `json:"sourceStore,omitempty"`
+	// TargetStore defines the specification of object store provider for storing backups.
+	// +required
+	TargetStore *StoreSpec `json:"targetStore,omitempty"`
+	// MaxBackupAge defines the maximum age in days that one snapshot must be in order to be copied.
+	// By default all snapshots will be copied.
+	// +optional
+	MaxBackupAge *uint32 `json:"maxBackupAge,omitempty"`
+	// MaxBackups defines how many snapshots will be copied starting with the most recent ones.
+	// +optional
+	MaxBackups *uint32 `json:"maxBackups,omitempty"`
+	// Image defines the etcd container image and tag
+	// +optional
+	Image *string `json:"image,omitempty"`
+}
+
+// EtcdCopyBackupsTaskStatus defines parameters that contains the current state of the etcd backup copy operation
+type EtcdCopyBackupsTaskStatus struct {
+	// ObservedGeneration is the most recent generation observed for this resource.
+	// +optional
+	ObservedGeneration *int64 `json:"observedGeneration,omitempty"`
+	// LastOperation represents the latest available observations of an etcd backup copy task current state.
+	// +optional
+	LastOperation *LastOperation `json:"lastOperation,omitempty"`
+	// TaskReference represents the details of the task that is currently deployed to preform the copy operation.
+	// +optional
+	TaskReference *corev1.TypedLocalObjectReference `json:"taskReference,omitempty"`
```

Comment:
I see this is still not addressed so let me justify my proposal. I think the name `TaskReference` is misleading since it's not a reference of a `Task`, but to a different object, e.g. `Job`, that has been created in order to perform whatever the task needs to perform.